tests/test_calc_error.py

```diff
@@ -0,0 +1,459 @@
+# Copyright © 2021 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration.  All Rights Reserved.
+
+import numpy as np
+import unittest
+
+from prog_models import *
+from prog_models.models import *
+
+class TestCalcError(unittest.TestCase):
+    """
+    Main Testing class for calc_error.
+
+    Validating that values are correctly being passed into the new calc_error calls and that we are receiving expected results!
+    """
+    def test_calc_error(self):
+        # Note, lowering time steps or increasing simulate threshold may cause this model to not run (takes too long)
+        m = BatteryElectroChemEOD()
+
+        options = {
+            'save_freq': 200, # Frequency at which results are saved
+            'dt': 1, # Time step
+        }
+
+        def future_loading(t, x=None):
+            if (t < 600):
+                i = 2
+            elif (t < 900):
+                i = 1
+            elif (t < 1800):
+                i = 4
+            elif (t < 3000):
+                i = 2
+            else:
+                i = 3
+            return m.InputContainer({'i': i})
+    
+        simulated_results = m.simulate_to(2000, future_loading, **options)
+
+        # Running calc_error before setting incorrect parameters
+        m.calc_error(simulated_results.times, simulated_results.inputs, simulated_results.outputs, dt=1)
```

Comment:
Since you don't save this result, I'm not sure it's necessary. This could be a place to save a little computation time